lib/core/src/Cardano/Pool/DB/Sqlite.hs

```diff
@@ -164,15 +171,33 @@ newDBLayer logConfig trace fp = do
                 [ StakeDistributionEpoch ==. fromIntegral epoch ]
                 []
 
-        , putStakePoolOwner = \poolId poolOwner ->
+        , putPoolRegistration = \PoolRegistrationCertificate
+            { poolId
+            , poolOwners
+            , poolMargin
+            , poolCost
+            } -> do
+            let poolMargin_ = fromIntegral $ fromEnum poolMargin
+            let poolCost_ = getQuantity poolCost
             repsert
+                (PoolMetadataKey poolId)
+                (PoolMetadata poolId poolMargin_ poolCost_)
+            forM_ poolOwners $ \poolOwner -> repsert
```

Comment:
**[question]**

`repsert` is used here. so "inserting" many rows at once and (a) if no entry then insert, (b) if entry present update it. Is it ok to have already present `poolId` and new `poolOwner` value? If no then shouldn't be there error? Moreover, don't we afraid of this `repsert` because of performance? So, we are updating many not changed rows every call?

Reply:
Hmmmm... So, there's nothing _wrong_ with replacing the value. However, thinking again about this well, it would probably be better to simply use `insert_` and make sure that certificates are rolled back correctly like the rest (which isn't the case right now, they are really just treated as "static" metadata, hence the repsert).